src/feature/dirauth/process_descs.c

```diff
@@ -174,27 +215,58 @@ dirserv_load_fingerprint_file(void)
   fingerprint_list_new = authdir_config_new();
 
   for (list=front; list; list=list->next) {
-    char digest_tmp[DIGEST_LEN];
     router_status_t add_status = 0;
     nickname = list->key; fingerprint = list->value;
     tor_strstrip(fingerprint, " "); /* remove spaces */
-    if (strlen(fingerprint) != HEX_DIGEST_LEN ||
-        base16_decode(digest_tmp, sizeof(digest_tmp),
-                      fingerprint, HEX_DIGEST_LEN) != sizeof(digest_tmp)) {
-      log_notice(LD_CONFIG,
-                 "Invalid fingerprint (nickname '%s', "
-                 "fingerprint %s). Skipping.",
-                 nickname, fingerprint);
-      continue;
-    }
+
+    /* Determine what we should do with the relay with the nickname field. */
     if (!strcasecmp(nickname, "!reject")) {
         add_status = FP_REJECT;
     } else if (!strcasecmp(nickname, "!badexit")) {
         add_status = FP_BADEXIT;
     } else if (!strcasecmp(nickname, "!invalid")) {
         add_status = FP_INVALID;
     }
-    add_fingerprint_to_dir(fingerprint, fingerprint_list_new, add_status);
+
+    /* Check fingerprint type by key length, and if it's RSA or ed25519. */
+    int is_ed25519 = (strlen(fingerprint) == BASE64_DIGEST256_LEN);
+    int is_rsa = (strlen(fingerprint) == HEX_DIGEST_LEN);
+    bool is_valid_fpr = true;
+
+    /* We only accept ed25519 or RSA keys */
+    if (!is_ed25519 && !is_rsa) {
+      is_valid_fpr = false;
+    }
+
+    tor_assert(is_ed25519 || is_rsa);
```

Comment:
This line will crash the authority without a useful warning if an approved_routers line is malformed.
Please don't do that.

Reply:
Instead, log a warning and continue with the next line in the file.